c2ctask/src/containerMapping.ts

```diff
@@ -0,0 +1,86 @@
+import tl = require('azure-pipelines-task-lib/task');
+import * as common from './common';
+import { IExecOptions } from 'azure-pipelines-task-lib/toolrunner';
+import { ICodeToCloud } from './c2c';
+
+
+/*
+* Class for Container Mapping functionality in Code to Cloud Decorator task.
+*/
+export class ContainerMapping implements ICodeToCloud {
+    private readonly taskType: common.TaskType;
+    private readonly version: string;
+    private readonly sectionDelim: string = ":::";
+    private readonly preJobStartTime: string = "PREJOBSTARTTIME";
+    private readonly imageOptions: IExecOptions = {
+        silent: true
+    };
+
+    constructor(inputString: string) {
+        this.taskType = inputString as common.TaskType;
+        this.version = common.getTaskVersion();
+    }
+
+    /*
+    * Set the start time of the job run.
+    */
+    private runPre() {
```

Comment:
Name of the function can be setJobStartTime

Reply:
I kept it consistent with the Decorators context (and the internal `taskType` parameter that we're using) to make it easier to understand the flow. I am going to keep it as is for now but might modify later if the design is changed.